### PR TITLE
Implented feed filter by title with regular expression

### PR DESCRIPTION
**Pull request by elegantsignal**
Additionally fixed build errors: `conversion from untyped int to string yields a string of one rune, not a string of digits (did you mean fmt.Sprint(x)?)` in  **telegram_test.go**

### CODE REVIEW

File: app/proc/processor.go [hunk 1]
```diff
@@ -105,6 +112,12 @@ func (p *Processor) feed(name, url, telegramChannel string, max int) {
 			continue
 		}
 
+		if (Filter{}) != filter {
```
Comment (umputun) [inline app/proc/processor.go:115]: this check is confusing, a little bit. First of all, why this condition reversed? `filter != Filter{}` feels better. Second - such a compare with struct better be implemented as `isEmpty` member func of the filter if you want to make it ready for future additions. 

Another solution is to make it FilterTitle and just compare the string for ""

File: app/proc/processor.go [hunk 2]
```diff
@@ -105,6 +112,12 @@ func (p *Processor) feed(name, url, telegramChannel string, max int) {
...
+			skip, _ := filter.skip(item)
```
Comment (umputun) [inline app/proc/processor.go:116]: ignoring error makes me nervous every time I see it. This is rarely justified and if it is - it needs a comment why. In this case, I see no reason to ignore

File: app/proc/processor.go [hunk 3]
```diff
@@ -142,3 +155,18 @@ func (p *Processor) setDefaults() {
 		p.Conf.System.UpdateInterval = time.Minute * 5
 	}
 }
+
+func (filter *Filter) skip(item feed.Item) (bool, error) {
+	if filter.Title != "" {
+		matched, err := regexp.MatchString(filter.Title, item.Title)
+		if err != nil {
+			log.Printf("[WARN] wrong title's regular expression: %s", filter.Title)
```
Comment (umputun) [inline app/proc/processor.go:163]: Error handling shouldn't be mixed like this. The func should handle it or by returning an error (with necessary context) or by "handling" it (log.Print in your case), but not both.

File: app/proc/processor_test.go [hunk 4]
```diff
@@ -27,3 +28,24 @@ func TestSetDefault(t *testing.T) {
 
 	assert.EqualValues(t, expectedConf.System, p.Conf.System)
 }
+
+func TestFilter(t *testing.T) {
+	feedItem := feed.Item{Title: "Foo Bar (Part 1)"}
+	filter := Filter{Title: "(Part \\d+)"}
+	result, _ := filter.skip(feedItem)
```
Comment (umputun) [inline app/proc/processor_test.go:35]: again, no error check even in the test. And also missing a test for the real error case

Comment (umputun) [reply app/proc/processor_test.go:35]: 3 tests you have here can be replaced with a single table test, like this one https://github.com/umputun/feed-master/blob/master/app/feed/parser_test.go#L33

File: app/proc/telegram_test.go [hunk 5]
```diff
@@ -64,7 +64,7 @@ func TestSendIfContentLengthZero(t *testing.T) {
 
 	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusInternalServerError)
-		w.Header().Set("Content-Length", string(0))
+		w.Header().Set("Content-Length", fmt.Sprint(0))
```
Comment (umputun) [inline app/proc/telegram_test.go:67]: I'm not sure why ` fmt.Sprint(0)` instead of `"0"`

File: app/proc/telegram_test.go [hunk 6]
```diff
@@ -174,7 +174,7 @@ func TestGetContentLengthNotFound(t *testing.T) {
 		t.Run(strconv.Itoa(i), func(t *testing.T) {
 			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 				w.WriteHeader(tc.statusCode)
-				w.Header().Set("Content-Length", string(tc.contentLength))
+				w.Header().Set("Content-Length", fmt.Sprint(tc.contentLength))
```
Comment (umputun) [inline app/proc/telegram_test.go:177]: here you probably want strconv.Itoa instead of fmt.Sprint

**Review by umputun — CHANGES_REQUESTED** [review]
thx for PR. Just a few small issues

File: app/proc/telegram_test.go [hunk 7]
```diff
@@ -218,13 +218,13 @@ func TestDownloadAudioIfRequestError(t *testing.T) {
...
 	assert.Nil(t, got)
-	assert.EqualError(t, err, fmt.Sprintf("Get %s: EOF", ts.URL))
+	assert.EqualError(t, err, fmt.Sprintf("Get %q: EOF", ts.URL))
 }
 
 func TestDownloadAudio(t *testing.T) {
 	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusOK)
-		w.Header().Set("Content-Length", string(4))
+		w.Header().Set("Content-Length", fmt.Sprint(4))
```
Comment (umputun) [inline app/proc/telegram_test.go:227]: looks like you missed this one

File: app/proc/store.go [hunk 8]
```diff
@@ -82,7 +82,7 @@ func (b BoltDB) Save(fmFeed string, item feed.Item) (bool, error) {
 }
 
 // Load from bold for given feed, up to max
-func (b BoltDB) Load(fmFeed string, max int) ([]feed.Item, error) {
+func (b BoltDB) Load(fmFeed string, max int, skipJunk bool) ([]feed.Item, error) {
```
Comment (umputun) [inline app/proc/store.go:85]: not sure I get it. Under what circumstances skipJunk false will make sense? I mean if we bothered to set up a filter it means we want to apply it.

Comment (elegantsignal) [reply app/proc/store.go:85]: We need to load "junks" for UI. Method `Load` serve both endpoints - rss and feed-page. For page we need to call them with `skipJunk==false`.

File: app/proc/processor.go [hunk 9]
```diff
@@ -105,6 +112,15 @@ func (p *Processor) feed(name, url, telegramChannel string, max int) {
 			continue
 		}
 
+		skip, err := filter.skip(item)
```
Comment (umputun) [inline app/proc/processor.go:115]: this part is confusing. The problem here is an implicit expectation of valid skip even if err != nil. This is unusual and even odd. I suggest adding to your `if skip` a condition `&& err ==nil`.

File: app/proc/store.go (hunk 8 above)
Comment (umputun) [reply app/proc/store.go:85]: ok, probably I'd do this logic (include/exclude) outside of the Load, but your way is ok too.

**Review by umputun — APPROVED** [review]

**Pull request merged**